src/els_parser.erl

```diff
@@ -402,3 +409,59 @@ poi(Pos, Kind, Id) ->
 poi(Pos, Kind, Id, Data) ->
   Range = els_range:range(Pos, Kind, Id, Data),
   els_poi:new(Range, Kind, Id, Data).
+
+%% @doc Fold over nodes in the tree
+%%
+%% Modified version of `erl_syntax_lib:fold/3', to get control over
+%% what subtrees should be folded over for certain types of nodes.
+-spec fold(fun((tree(), term()) -> term()), term(), tree()) -> term().
+fold(F, S, Tree) ->
+  case subtrees(Tree, erl_syntax:type(Tree)) of
+    [] -> F(Tree, S);
+    Gs -> F(Tree, fold_1(F, S, Gs))
+  end.
+
+-spec fold_1(fun((tree(), term()) -> term()), term(), [[tree()]]) ->
+  term().
+fold_1(F, S, [L | Ls]) ->
+  fold_1(F, fold_2(F, S, L), Ls);
+fold_1(_, S, []) ->
+  S.
+
+-spec fold_2(fun((tree(), term()) -> term()), term(), [tree()]) ->
+  term().
+fold_2(F, S, [T | Ts]) ->
+  fold_2(F, fold(F, S, T), Ts);
+fold_2(_, S, []) ->
+  S.
+
+-spec subtrees(tree(), atom()) -> [[tree()]].
```

Comment:
The most important change in this PR is in this function which filters out the nodes that might contains a syntax tree node of type `atom` which we don't want as a POI.